cumulus/test/service/src/lib.rs

```diff
@@ -301,7 +301,7 @@ impl TestNodeBuilder {
     }
 
     /// Build the [`TestNode`].
-    pub async fn build(self, role: Role) -> TestNode {
+    pub async fn build(self, role: Role, force_authoring: bool, force_synced: bool) -> TestNode {
```

Comment:
These parameters combined already don't sound super easy to understand what's absolutely necessary, better to make it explicitly tell that they are used in the primary node config.

```suggestion
    pub async fn build(self, role: Role, primary_force_authoring: bool, primary_force_synced: bool) -> TestNode {
```

I kind of feel the tests may be fixed with smaller changes, but I think they are good for now.